app/scripts/metamask-controller.js

```diff
@@ -5878,7 +5878,11 @@ export default class MetamaskController extends EventEmitter {
         },
         // network configuration-related
         setActiveNetwork: async (networkClientId) => {
-          await this.networkController.setActiveNetwork(networkClientId);
+          // Skip setting the globally selected network if there are
+          // pending approvals otherwise they will get inadvertently cleared
+          if (this.approvalController.getTotalApprovalCount() === 0) {
```

Comment:
Should we log or even somehow inform the user in case this skips (`else`)?

Reply:
I don't think we need to. The only way the user will notice the globally selected network mismatched from the dapp selected network is if they interact with the approvals via the popup instead of the notification window. If they use the popup to interact with the approvals, they will end up on the homepage view and see the mistmatched globally selected network afterwards. If they use the notification window to interact with the approvals, the window will close itself after all approvals are handled, and so the user will not see the globally selected network was mismatched since they never hit the homepage view and if they try to open it via the popup, extension will repoint the globally selected network at that time.